pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/api.go

```diff
@@ -104,25 +113,42 @@ func (s *apiScaffolder) Scaffold() error {
 		return fmt.Errorf("error updating config/samples: %v", err)
 	}
 
-	controller := &controllers.Controller{
-		ControllerRuntimeVersion: golangv3scaffolds.ControllerRuntimeVersion,
-	}
-	if err := scaffold.Execute(
-		controller,
-	); err != nil {
-		return fmt.Errorf("error scaffolding controller: %v", err)
-	}
+	if isLegacyLayout {
+		controller := &controllers.Controller{
+			ControllerRuntimeVersion: golangv3scaffolds.ControllerRuntimeVersion,
+			IsLegacyLayout:           true,
+		}
+		if err := scaffold.Execute(
+			controller,
+		); err != nil {
+			return fmt.Errorf("error scaffolding controller: %v", err)
+		}
+
+		if err := s.updateControllerCode(*controller); err != nil {
+			return fmt.Errorf("error updating controller: %v", err)
+		}
 
-	if err := s.updateControllerCode(*controller); err != nil {
-		return fmt.Errorf("error updating controller: %v", err)
+	} else {
+		controller := &controllers.Controller{
+			ControllerRuntimeVersion: golangv4scaffolds.ControllerRuntimeVersion,
+			IsLegacyLayout:           false,
+		}
+		if err := scaffold.Execute(
+			controller,
+		); err != nil {
+			return fmt.Errorf("error scaffolding controller: %v", err)
+		}
+		if err := s.updateControllerCode(*controller); err != nil {
+			return fmt.Errorf("error updating controller: %v", err)
+		}
```

Comment:
Maybe we can prevent some code duplication with something like:
```go
controller := &controllers.Controller{
    ControllerRuntimeVersion: golangv3scaffolds.ControllerRuntimeVersion,
    IsLegacyLayout:           true,
}

if !isLegacyLayout {
    controller.ControllerRuntimeVersion = golangv4scaffolds.ControllerRuntimeVersion
    controller.IsLegacyLayout = false
}

if err := scaffold.Execute(controller); err != nil {
      return fmt.Errorf("error scaffolding controller: %v", err)
}

if err := s.updateControllerCode(*controller); err != nil {
	return fmt.Errorf("error updating controller: %v", err)
}
```

Reply:
done